src/app/Mappings/components/MappingBuilder/MappingTargetSelect.tsx

```diff
@@ -0,0 +1,64 @@
+import * as React from 'react';
+import SimpleSelect, { OptionWithValue } from '@app/common/components/SimpleSelect';
+import { ICNVNetwork } from '@app/Providers/types';
+import { MappingTarget, MappingType } from '@app/Mappings/types';
+import { IMappingBuilderItem } from './MappingBuilder';
+
+interface IMappingTargetSelectProps {
+  id: string;
+  mappingType: MappingType;
+  builderItems: IMappingBuilderItem[];
+  itemIndex: number;
+  setBuilderItems: (items: IMappingBuilderItem[]) => void;
+  availableTargets: MappingTarget[];
+  placeholderText: string;
+}
+
+const MappingTargetSelect: React.FunctionComponent<IMappingTargetSelectProps> = ({
+  id,
+  mappingType,
+  builderItems,
+  itemIndex,
+  setBuilderItems,
+  availableTargets,
+  placeholderText,
+}: IMappingTargetSelectProps) => {
+  const setTarget = (target: MappingTarget) => {
+    const newItems = [...builderItems];
+    newItems[itemIndex] = { ...builderItems[itemIndex], target };
+    setBuilderItems(newItems);
+  };
+
+  const targetOptions: OptionWithValue<MappingTarget>[] = availableTargets.map((target) => ({
+    value: target,
+    toString: () => {
+      if (mappingType === MappingType.Network) {
+        return (target as ICNVNetwork).name;
+      }
+      if (mappingType === MappingType.Storage) {
+        return target as string;
+      }
+      return '';
```

Comment:
I wonder if there is a cleaner way to code this? I have used a similar pattern in the past but tried recently to switch to ternary expressions for these. It would probably be overkill for a switch statement. This works fine but I am always wishing there was some sort of middle ground.

Reply:
Yeah, I was thinking of a ternary expression, but that requires choosing either network or storage as the condition and then having "other" be assumed to be the other. i.e. if it's a network mapping do this, else do this. I don't know if we'll ever need to add a third kind of mapping, but this allows for that. Could always refactor that later, so if you think it should be a ternary I'm fine with that.

Reply:
I am fine with if statements if there are more than 3 statements. & maybe we switch to switch statements if there are more than 3. Its more of a gripe with JS than with your PR :)